fix/update to fetch holiday-data up to 2000 results

inkybr opened the pull request:
PR #5 doesn't provide the data of whole year of 2021. This PR attempts to add rest of them and fix #4 .

Data source is changed to the Open Data API of https://data.taipei . Because the api has a limit of 1000 result, it will be capped at 2021/4/23. So a change needs to made to parallelly fetch different segments of data and merge them to retrieve a complete set of holiday data.

For now, it only retrieves the first 2000 results. The api has only 1077 results as for now, so I figure 2000 may be enough until a more refined solution is made.

inkybr committed:
Update taiwan-holiday-cache.json to 2021

data from https://data.taipei/api/v1/dataset/b0011e96-3fc3-43ec-8bf5-07fb46dd22bb?scope=resourceAquire

Commit:
update holiday-data fetch method

Data source is changed to the Open Data API of https://data.taipei . Because the api has a limit of 1000 result, the data will be capped at 2021/4/23. So a change need to made to parallelly fetch different segments of data and merge them to retrieve the complete holiday datas.

For now, it only retrieves the first 2000 data. The api has only 1077 results, so I figure 2000 may be enough until a more refined solution is made.

bcgodev commented:
This PR is related to another PR #5 and issue #4

Commit:
refactor(common): refactor taiwan-holiday.js for cleaner code

ensky commented on lib/opendata-tw.js:
```diff
@@ -23,6 +23,7 @@ class OpendataTW {
 // TODO: fetch method rather than url
-OpendataTW.url = 'https://data.ntpc.gov.tw/api/datasets/308DCD75-6434-45BC-A95F-584DA4FED251/json?size=2000';
+OpendataTW.urlFirst1000 = 'https://data.taipei/api/v1/dataset/b0011e96-3fc3-43ec-8bf5-07fb46dd22bb?scope=resourceAquire&q=&limit=1000=&offset=0';
+OpendataTW.urlSecond1000 = "https://data.taipei/api/v1/dataset/b0011e96-3fc3-43ec-8bf5-07fb46dd22bb?scope=resourceAquire&q=&limit=1000=&offset=1000";
```
```suggestion
OpendataTW.urlSecond1000 = 'https://data.taipei/api/v1/dataset/b0011e96-3fc3-43ec-8bf5-07fb46dd22bb?scope=resourceAquire&q=&limit=1000=&offset=1000';
```

for quote consistency

inkybr replied:
Great suggestion. I should notice the code style consistency and readability. I'll make another patch soon.

inkybr replied:
I've added a patch to implement the suggestion. Should we continue the discussion of fetching method in issue #9 and implement it in other PR?

ensky reviewed (commented):
Thanks for the PR, nice work!

However, if 1000 results roughly means ~8 years of holidays, I think maybe we should check if there's another pages exists. Otherwise this library might needs to change at roughly 2029.

Check with result.count maybe a good idea?

Commit:
style(OpendataTW): use single quote instead of double quote for coding style consistency

Commit:
Solves conflicts with upstream

Merge branch 'master' of https://github.com/ensky/taiwan-holiday.js into fix/update-data-and-cache

# Conflicts:
#	data/taiwan-holiday-cache.json
#	lib/taiwan-holiday.js
